packages/ethers/src/index.ts

```diff
@@ -199,6 +234,37 @@ export class TurnkeySigner extends ethers.Signer {
       activityType: type,
     });
   }
+
+  async signTypedData(
+    domain: TypedDataDomain,
+    types: Record<string, Array<TypedDataField>>,
+    value: Record<string, any>
+  ): Promise<string> {
+    // Populate any ENS names
+    const populated = await ethers.utils._TypedDataEncoder.resolveNames(
+      domain,
+      types,
+      value,
+      async (name: string) => {
+        assertNonNull(this.provider);
+
+        const address = await this.provider?.resolveName(name);
+        assertNonNull(address);
+
+        return address ?? "";
+      }
+    );
+
+    return this._signMessageWithErrorWrapping(
+      ethers.utils._TypedDataEncoder.hash(
+        populated.domain,
+        types,
+        populated.value
+      )
+    );
+  }
+
+  _signTypedData = this.signTypedData.bind(this);
```

Comment:
While most clients use `signTypedData`, some (like Gnosis SDK) expect to see `_signTypedData` (see https://github.com/safe-global/safe-core-sdk/blob/7959821ab08c96cf3babb9ed906c01d644ac49f4/packages/protocol-kit/src/adapters/ethers/utils/index.ts#L21-L22)